### DISCUSSION TITLE
Support for "row" field on auxiliaryFields?

### DISCUSSION THREAD

**Discussion by danawoodman** [alexandercerutti/passkit-generator#115] [Q&A] [answered]
From what I can tell, passkit-generator should work with the `row` property ([See](https://developer.apple.com/forums/thread/108302)) but I cannot figure out how. If I do the following:

```ts
const pass = new PKPass(...)
pass.type = 'eventTicket'
pass.auxiliaryFields.push(
  { value: 'Hello', label: 'Foo', key: 'foo', row: 0 },
  { value: 'World', label: 'Bar', key: 'bar', row: 1 },
)
```

I get a TS error and a Joi validation error:

```
ValidationError happened. Cannot add field. "row" is not allowed
[Error [ValidationError]: "row" is not allowed] {
  _original: { key: 'foo', value: 'Hello', label: 'Foo', row: 0 },
  details: [
    {
      message: '"row" is not allowed',
      path: [Array],
      type: 'object.unknown',
      context: [Object]
    }
  ]
}
ValidationError happened. Cannot add field. "row" is not allowed
```

The type for `pass.auxillaryFields` is `Field` but in the typing I see it is `PassField`: https://github.com/alexandercerutti/passkit-generator/blob/267d8a6e8c52337f94b7ba75359f8d3cb10a26ae/src/schemas/PassFields.ts#L16 

How can I add a `row` attribute to my auxiliary fields?

**Comment by alexandercerutti** [comment]
Hey @danawoodman, thanks for using passkit-generator!

Let me start by saying that I never understood how to use the `row` attribute. I don't know how it gets rendered and the examples on Apple Developers were non-existing. So, I never tested it thoroughly, and no one ever reported issues.

That said, I made a bit of debugging. I found out where the issue lies.
The schema that gets validated doesn't actually contain the `row` [because it is the "raw" one](https://github.com/alexandercerutti/passkit-generator/blob/master/src/FieldsArray.ts#L78).

I think that I should create a new schema just for Fields with row and validate against that if we match the current type to be an `auxiliaryField`.

Would you be willing to test a solution for me?

Let me know!

**Reply by danawoodman** [reply]
Hi @aaliakseyenka thanks for the quick reply! I'd love to test that out, as our design requires multiple lines of data and the auxiliary row field is exactly what we're looking for :) Let me know when you want a test

**Reply by alexandercerutti** [reply]
@danawoodman you tagged the wrong person ahaha but thank you! I'm performing myself a few tests. And then I'll open a PR and tag you!

**Reply by danawoodman** [reply]
@alexandercerutti whoops, Github autocorrect to the wrong person. Excited to play around with `row`

**Comment by alexandercerutti** [comment, marked as answer]
Fix released in v3.1.7!
